app/api/orders.py

```diff
@@ -188,20 +188,50 @@ def before_get_object(self, view_kwargs):
 
     def before_update_object(self, order, data, view_kwargs):
         """
+        before update object method of order details
+        1. admin can update all the fields.
+        2. event organizer
+            a. own orders: he/she can update selected fields.
+            b. other's orders: can only update the status that too when the order mode is free. No refund system.
+        3. order user can update selected fields of his/her order when the status is pending.
+        The selected fields mentioned above can be taken from get_updatable_fields method from order model.
         :param order:
         :param data:
         :param view_kwargs:
         :return:
         """
-        # Admin can update all the fields while Co-organizer can update only the status
-        if not has_access('is_admin'):
-            for element in data:
-                if element != 'status':
-                    setattr(data, element, getattr(order, element))
+        if (not has_access('is_coorganizer', event_id=order.event_id)) and (not current_user.id == order.user_id):
+            raise ForbiddenException({'pointer': ''}, "Access Forbidden")
 
-        if not has_access('is_coorganizer', event_id=order.event.id):
-            raise ForbiddenException({'pointer': 'data/status'},
-                                     "To update status minimum Co-organizer access required")
+        if has_access('is_coorganizer_but_not_admin', event_id=order.event_id):
+            if current_user.id == order.user_id:
+                # Order created from the tickets tab.
+                for element in data:
```

Comment:
You can have some parts of this as a helper function (subroutine)? That way you can use it below as well. Would that be a good idea or an overkill?

Reply:
We could but I feel like if we have it like this then it is great for the readability of the code. You can just read the docstring and read the code and everything will be clear to you. Not sure this would be the case if I encapsulate it into a helper function.